cmake/GzSetCompilerFlags.cmake

```diff
@@ -342,6 +346,29 @@ macro(_gz_setup_msvc)
     # TODO: What flags should be set for PROFILE and COVERAGE build types?
     #       Is it even possible to generate those build types on Windows?
 
+    # Replace any higher warnings with our selected level
```

Comment:
I would prefer not to mess with the `CMAKE_{C/CXX}_` variables if possible (or not to do it more than we are currently doing). Would it work something like `add_compile_options(${MSVC_WARNING_LEVEL})`?

Reply:
Yeah, in general, I agree.  The issue here is that MSVC is really dumb about how it parses command line options, and if you have multiple warning switches, they can interact poorly.  Simply appending a new warning level to the end causes warnings to show up on the command line (iirc)

Reply:
The warning level bit is actually separate from the caching support here.  It's just to suppress that warning with MSVC

Reply:
> The warning level bit is actually separate from the caching support here. It's just to suppress that warning with MSVC

The comment was looking to avoid messing with `CMAKE_{C/CXX}_` for the warnings and for the optimization in the lines after. If you have already tested and suffer the multiple compile line options, I'm fine just by adding a comment before the `CMAKE_{C/CXX}_` lines explaining why we are not using`add_compile_options`.